### PR TITLE
(DOCSP-17889): Enhancements and gotchas for automatic GitHub deploy

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-17887
- https://jira.mongodb.org/browse/DOCSP-17889
- https://jira.mongodb.org/browse/DOCSP-18318

### Staged Changes (Requires MongoDB Corp SSO)

New sections in the Deploy Automatically with GitHub page corresponding to the various Jira tickets (and added/updated Summary bullets):

- [Make Changes from the UI -> DOCSP-17787](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-17889/deploy/deploy-automatically-with-github/#make-changes-from-the-ui)
- [Avoid Making Changes from the CLI-> DOCSP-18313](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-17889/deploy/deploy-automatically-with-github/#avoid-making-changes-from-the-cli)

New page:
- [Rollback Deployments -> DOCSP-17789](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-17889/deploy/rollback-deployment/)

Added this new page to the ToC. Also mentioned rollbacks and updated the number of deployments in the [Deployment History section of the Deploy page](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-17889/deploy/#deployment-history).

New warning requested in DOCSP-18313:
- [Deploy with the Realm CLI](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-17889/deploy/deploy-cli/#procedure)

DOCSP-18313 also requests adding a warning to the [Realm CLI Push](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-17889/cli/realm-cli-push/) page, but this PR does not incorporate that. That page is now auto-generated from the Realm CLI source, so any warning we add in `docs-realm` would be overwritten the next time the CLI docs are generated.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/includes/steps-github-deploy-rollback.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  :guilabel:`History` tab.
+
+  You'll see the 25 most recent application deployments. The 
+  :guilabel:`Change Origin` column shows whether the deploy originated from
+  changes in the UI, the linked GitHub repository, the Admin API, or the CLI.
+
+  .. figure:: /images/realm-app-deploy-history.png
+     :alt: Deployment History screen with Realm, Deployment, and History tabs circled
+---
+title: Re-Deploy a Prior Deploy
```
Comment (nlarew) [inline source/includes/steps-github-deploy-rollback.yaml:17]:
```suggestion
title: Re-Deploy a Prior Deployment
```

nitpick - alternatively we could use "Version". We also need to update the corresponding ref on the next line to match any changes here

File: source/includes/steps-github-deploy-rollback.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ref: re-deploy-prior-deploy
+content: |
+  Once you have selected a prior version to rollback to, press the
```
Comment (nlarew) [inline source/includes/steps-github-deploy-rollback.yaml:20]:
```suggestion
  Select a prior version to rollback to and then press the 
```

nit: technically we haven't told them to pick one yet!

File: source/deploy/deploy-automatically-with-github.txt [hunk 3]
```diff
@@ -39,20 +39,83 @@ Prerequisites
...
+.. _deploy-github-enable:
+
+Enable Automatic Deployment with GitHub
+---------------------------------------
 
 .. include:: /includes/steps/github-deploy.rst
 
+.. _deploy-github-rollback-to-prior-deploy:
+
+Rollback to a Previous Deploy
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/deploy/deploy-automatically-with-github.txt:51]: Hmm, fair. I've got an outstanding question related to this, so I'll come back to where this should live after that. I could break it out to a separate page if that makes more sense - I may do that depending on what I find out.

[the remark this answers is not in the document]
Comment (dacharyc) [reply source/deploy/deploy-automatically-with-github.txt]: Alright, Sumedha confirms you should be able to roll back to any deployment (I made some incorrect assumptions about this due to a current bug) so I've broken out the rollback procedure to its own page: [Rollback Deployments](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-17889/deploy/rollback-deployment/)

And have added it to the ToC and a mention of it in the [Deployment History section of the Deploy page](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-17889/deploy/#deployment-history).

File: source/deploy.txt [hunk 4]
```diff
@@ -192,10 +193,14 @@ changes.
 .. figure:: /images/deployment-history-table.png
    :alt: The table of application deployment history logs in the {+ui+}
 
+The :guilabel:`Deployment History` is also where you can :guilabel:`Export`
+your {+app+}, or use the :guilabel:`Re-Deploy` button to rollback to a prior 
+version.
```
Comment (nlarew) [inline source/deploy.txt:198]: Can we:
* move this above the screenshot
* link to the new rollback page

**Review by nlarew — APPROVED** [review]
LGTM % one more comment

**Pull request merged**